src/AWS.Deploy.CLI/Commands/CommandFactory.cs

```diff
@@ -209,8 +209,8 @@ private Command BuildDeployCommand()
                         deploymentSettings = await _deploymentSettingsHandler.ReadSettings(applyPath);
                     }
 
-                    var awsCredentials = await _awsUtilities.ResolveAWSCredentials(input.Profile ?? deploymentSettings?.AWSProfile);
-                    var awsRegion = _awsUtilities.ResolveAWSRegion(input.Region ?? deploymentSettings?.AWSRegion);
+                    var (awsCredentials, regionFromProfile) = await _awsUtilities.ResolveAWSCredentials(input.Profile ?? deploymentSettings?.AWSProfile);
+                    var awsRegion = _awsUtilities.ResolveAWSRegion((input.Region ?? deploymentSettings?.AWSRegion) ?? regionFromProfile);
```

Comment:
nit: you shouldn't need parentheses around  `input.Region ?? deploymentSettings?.AWSRegion`. You can just do:
```
input.Region ?? deploymentSettings?.AWSRegion ?? regionFromProfile
```